content/general/authentication.textile

```diff
@@ -139,22 +141,91 @@ ably.requestToken({ clientId: 'client@example.com' }, function(err, token) {
+This is similar to signing an "Ably TokenRequest":/realtime/authentication#request-token, but the client does not need then to request an Ably Token, instead being able to use the Ably JWT as a token in itself. Any compliant third-party JWT library may be used to create the JWT without requiring the token to be issued by Ably. This can be useful for situations where an Ably client library is not available, such as an embedded device connecting to Ably via MQTT.
 
-"Tokens":#tokens are issued from an existing "API key":https://support.ably.io/solution/articles/3000030054-what-is-an-app-api-key, and their capabilities can, at most, match the capabilities of the issuing API key. If an API key must be shared with a third party, then it is recommended that "the principle of least privilege":http://en.wikipedia.org/wiki/Principle_of_least_privilege is considered assigning only the capabilities needed by that third party. Thus, any Ably requests authenticated using that API key or tokens issued from that API key, will be restricted to the capabilities assigned to the key.
+<a href="/images/diagrams/ably-jwt-auth.png" target="_blank">
+  <img src="/images/diagrams/ably-jwt-auth.png" style="width: 100%" alt="Ably JWT auth method">
+</a>
 
-Capabilities for tokens are determined as follows:
+An example of creating an Ably JWT can be seen below:
+
+bc[javascript]. var header = { 
+    "typ":"JWT",
+    "alg":"HS256",
+    "kid": "{{API_KEY_NAME}}"
+  }
+  var claims = {
+    "iat": currentTime,
+    "exp": currentTime + 3600,
+    "x-ably-capability": "{\"*\":[\"*\"]}"
+  }
+  var base64Header = btoa(header);
+  var base64Claims = btoa(claims);
+  /* Apply the hash specified in the header */
+  var signature = hash((base64Header + "." + base64Claims), SECRET);
+  var ablyJwt = base64Header + "." + base64Claims + "." + signature;
+
+*Note:* At present Ably does not support asymmetric signatures based on a keypair belonging to a third party. If this is something you'd be interested in, please "get in touch":https://www.ably.io/contact.
+
+h4(#jwt-embed-process). Ably-compatible token is embedded in a External JWT from your server and passed to clients
+
+If a system has an existing "JWT":https://jwt.io/ scheme, it can be useful to embed an Ably-compatible token as a claim within it. To do this, the External JWT and need to ensure it meets the following requirements:
```

Comment:
Sentence "To do this, the External JWT and need to ensure it meets the following requirements" is missing something?